docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,777 @@
+
+
+###### Additional remarks on the system model
+We explicitly overlook here two assumptions of synchrony that may be necessary for achieving liveness of **other** protocols, but they are not necessary within the ICS3 protocol.
+These assumptions are: (i) partial synchrony is necessary for achieving liveness of a chain if that chain is implemented using a deterministic consensus protocol (e.g., Tendermint Core); (ii) synchrony is necessary for guaranteeing message delivery in any connection-oriented protocol such as TCP, which is most likely the underlying network transport layer (implementing the `CommunicationLayer` between modules and relayers).
+
+
+#### Properties
+
+###### Preliminary abstractions
+We introduce several abstractions that are relevant at this point:
+
+1. A `ConnectionParty` is a data type encapsulating the details that identify a party.
+
+```golang
+type ConnectionParty interface {
+  connectionIdentifier    Identifier
+  clientIdentifier        Identifier
+  prefix                  Prefix
+}
+```
+
+2. A `ConnectionState` captures the state of a connection, and may have one of the following values.
+
+```typescript
+enum ConnectionState {
+  INIT,
+  TRYOPEN,
+  OPEN,
+}
+```
+
+3. A `ConnectionEnd` is a data type that captures all the details of a connection at a party.
+One of the goals of the connection handshake protocol is to allocate an object of this type at each of Alice and Bob.
+
+```golang
+type ConnectionEnd interface {
+  state                             ConnectionState
+  remoteParty                       ConnectionParty
+  localParty                        ConnectionParty
+  version                           string | []string
+}
+```
+
+Note that the definition of `ConnectionEnd` includes a _remote_ and a _local_ field; in a specific `ConnectionEnd` object, the local party is the one storing this object, and the remote party is the other one.
+
+
+4. A `Proof` is a data type that abstracts over the details of a cryptographic proof that any module can create, and another module can verify.
+Intuitively, a proof is helpful for some module to guarantee that it is in a certain state.
+
+```golang
+type Proof struct {
+  content     CommitmentProof // the actual proof
+  height      uint64
+}
+
+```
+
+Further, we also define the interface of a `Store` on each chain, which maintains the state of the chain.
+This is a straightforward key/value store.
+
+
+```golang
+type Store interface {
+  // Returns the value stored at key `k`, possibly returning null if no value
+  // was previously set at key `k`.
+  get(k) -> byte[]
+
+  // Sets the value at position `k` to be `v`, overwritting if this key is
+  // already present; always returns null (success).
+  set(k string, v byte[]) -> null
+}
+```
+
+Importantly, this store is shared with the relayer.
```

Comment:
What does this mean? The relayer does not have direct access to the store, only via queries. Queries can be made by anyone, relayer, user, scripts, etc.

Reply:
From the point of view of this spec it does not seem relevant that users or scripts (in addition to relayers) can make these queries. I will add a side-note, however, to mention this detail if you believe it would help.

Reply:
"shared" might imply that the relayer can write there. There is no special access that the relayer requires as compared with other processes. The store is queryable by anyone. Maybe something like:
```suggestion
Importantly, queries to this store can be made.
```

Reply:
I see your point, agree.